Documented function parameters for get_serializer() are misleading

The documentation for [GenericAPIView](http://www.django-rest-framework.org/api-guide/generic-views/#genericapiview) shows that the function signature for `get_serializer` is `get_serializer(self, instance=None, data=None, files=None, many=False, partial=False, allow_add_remove=False)`.

This implies that it should be possible to do something like this:
```python
class TestViewSet(viewsets.ModelViewSet):
    serializer_class = TestModelSerializer
    def get_serializer(self, instance=None, data=None, files=None, many=False,  
                       partial=False, allow_add_remove=False):                     
        # Do custom stuff here                                                     
        return super(TestViewSet, self).get_serializer(instance=instance, data=data,
                                                       files=files, many=many, partial=partial,
                                                       allow_add_remove=allow_add_remove)
```
However, this fails with `TypeError: __init__() got an unexpected keyword argument 'files'` (and the same message for `allow_add_remove` if `files` is removed).

If those two parameters are removed, list and detail endpoints fail with `AssertionError: When a serializer is passed a `data` keyword argument you must call `.is_valid()` before attempting to access the serialized `.data` representation.` This is because the ListModelMixin and RetrieveModelMixin calls `self.get_serializer(instance)` rather than `self.get_serializer(instance=instance)`.

I'm not exactly sure what the best resolution for this is; I expect that `allow_add_remove` and `files` may be used in other places such that removing them from the documentation would be inappropriate. However, it seems that the `GenericAPIView` should allow subclasses to call e.g. `serializer = self.get_serializer(files=files)` without raising errors, which doesn't appear to be the case currently.

Similarly, the AssertionError above is surprising since there is no indication in the documentation that `get_serializer` will be called with an object instance as an argument rather than a keyword argument.

--- a/docs/api-guide/generic-views.md
+++ b/docs/api-guide/generic-views.md
@@ -195,7 +195,7 @@ These override points are also particularly useful for adding behavior that occu
 You won't typically need to override the following methods, although you might need to call into them if you're writing custom views using `GenericAPIView`.
 
 * `get_serializer_context(self)` - Returns a dictionary containing any extra context that should be supplied to the serializer.  Defaults to including `'request'`, `'view'` and `'format'` keys.
-* `get_serializer(self, instance=None, data=None, files=None, many=False, partial=False, allow_add_remove=False)` - Returns a serializer instance.
+* `get_serializer(self, instance=None, data=None, many=False, partial=False)` - Returns a serializer instance.
 * `get_paginated_response(self, data)` - Returns a paginated style `Response` object.
 * `paginate_queryset(self, queryset)` - Paginate a queryset if required, either returning a page object, or `None` if pagination is not configured for this view.
 * `filter_queryset(self, queryset)` - Given a queryset, filter it with whichever filter backends are in use, returning a new queryset.

